backend/forecastbox/config.py

```diff
@@ -157,12 +166,53 @@ class FIABConfig(BaseSettings):
     model_config = SettingsConfigDict(env_file=".env", env_nested_delimiter="__", env_prefix="fiab__")
 
     general: GeneralSettings = Field(default_factory=GeneralSettings)
+    product: ProductSettings = Field(default_factory=ProductSettings, description="Product specific settings")
+
     auth: AuthSettings = Field(default_factory=AuthSettings)
 
     db: DatabaseSettings = Field(default_factory=DatabaseSettings)
     api: BackendAPISettings = Field(default_factory=BackendAPISettings)
     cascade: CascadeSettings = Field(default_factory=CascadeSettings)
 
+    @classmethod
+    def settings_customise_sources(
+        cls,
+        settings_cls: type[BaseSettings],
+        init_settings: PydanticBaseSettingsSource,
+        env_settings: PydanticBaseSettingsSource,
+        dotenv_settings: PydanticBaseSettingsSource,
+        file_secret_settings: PydanticBaseSettingsSource,
+    ) -> tuple[PydanticBaseSettingsSource, ...]:
+        return env_settings, file_secret_settings, dotenv_settings, TomlConfigSettingsSource(settings_cls, fiab_home / "config.toml"), init_settings
+
+    def _get_toml(self, skip_secrets: bool = True, **k) -> str:
+        json_config = self.model_dump(mode="json", **k)
+
+        if skip_secrets:
+            (json_config.get('auth', {}).get('oidc', {}) or {}).pop('client_secret', None)
+            json_config.get('auth', {}).pop('jwt_secret', None)
+
+        toml_config = toml.dumps(json_config)
+        return toml_config
+
+    def save_to_file(self) -> None:
+        """Save current configuration to toml file"""
+
+        config_path = fiab_home / "config.toml"
+        config_path.parent.mkdir(parents=True, exist_ok=True)
+        with open(config_path, 'w') as f:
+            f.write(self._get_toml(exclude_defaults = True, exclude_none = True))
+
+    def save_example_to_file(self) -> None:
+        """Save example configuration to toml file"""
+
+        config_path = fiab_home / "config.toml.example"
+        config_path.parent.mkdir(parents=True, exist_ok=True)
+        with open(config_path, 'w') as f:
+            f.write(self._get_toml())
+
+    def __del__(self) -> None:
+        self.save_to_file()
```

Comment:
this also feels like a very heavy handed thing... could lead to some weird situations, like shutting down mid-initialization possibly destroys your config file